src/Http/GraphRequestUtil.php

```diff
@@ -30,16 +30,27 @@ class GraphRequestUtil
      * @param string $baseUrl if empty, is overwritten with $client's national cloud
      * @param string $endpoint can be a full URL
      * @param string $apiVersion
-     * @return UriInterface|null
+     * @return UriInterface
+     * @throws \InvalidArgumentException
      */
-    public static function getRequestUri(string $baseUrl, string $endpoint, string $apiVersion = "v1.0"): ?UriInterface {
-        // If endpoint is a full url, ensure the host is a national cloud or custom host
+    public static function getRequestUri(string $baseUrl, string $endpoint, string $apiVersion = "v1.0"): UriInterface {
+        // If endpoint is a full url, ensure the host is a national cloud
         if (parse_url($endpoint, PHP_URL_SCHEME)) {
-            return (NationalCloud::containsNationalCloudHost($endpoint)) ? new Uri($endpoint) : null;
+            if (NationalCloud::containsNationalCloudHost($endpoint)) {
```

Comment:
My only concern here is that as new national (or even state-level) clouds come on board, we will have to keep updating NationalCloud.php. Now mind you, this hasn't happened often, yet, so perhaps this is good value for the customer.

As it is now, if they get the cloud wrong, they will get an error along the lines that the host name doesn't resolve. That will be a decent hint for them to check the cloud.

Reply:
Interesting pointer. I agree that this adds a burden on us to always be up to date & we'd potentially break a customer's experience.

I am happy with removing this restriction and allowing any valid full endpoint to work.

Worst case (without this validation) is the host name resolution error which would still prompt a user to double check their hostname

Reply:
Which leads me to need to update [this](https://github.com/microsoftgraph/msgraph-sdk-php-core/pull/5/files/9fbe9e9a2290e550cd267cad63741de4e65a2cde#diff-849151e7b2ea35fbfeca79419f39a80ebc43ed0c66cd50f309db361f7682c4a9R347) to check the full request URI to prevent sending tokens to non-graph endpoints

Reply:
Removed extra validation + updated add headers logic